### ISSUE TITLE
Allow gzip content encoding for imports.

### ISSUE THREAD

**Issue by naturalethic** [typesense/typesense#580]
Please allow a gzipped content encoding for imports.  Further, why not for all apis?

**Comment by kishorenc** [comment]
This is a good idea, but web server support for compressed request bodies is rather slim, and that's the case for the http library that Typesense is using as well. So, we will have to build a custom compression/decompression layer which is not something that's trivial to just turn ON, as I had hoped.

**Comment by nojvek** [comment]
Is this something that could be a community contribution. Seems like waste of bandwidth to send uncompressed imports. Would probably get a Perf boost as well.

**Comment by kishorenc** [comment]
@nojvek 

I agree. We want to improve this but also happy to accept PRs. The change will require parsing the request stream as a gzip stream so that the request body is decoded before being buffered as string here: https://github.com/typesense/typesense/blob/v0.24-nested/src/http_server.cpp#L583
